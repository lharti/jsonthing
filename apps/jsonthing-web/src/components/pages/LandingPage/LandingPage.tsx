import { CreateNewDocButton } from '@/components/CreateNewDocButton'
import Image from 'next/image'
import React from 'react'

export const LandingPage: React.FC = ({}) => {
    return (
        <div className="mx-auto max-w-screen-xl px-8 pt-32">
            <main
                className={`
                  flex flex-col items-center gap-8

                  lg:flex-row lg:items-start lg:justify-between
                `}
            >
                <div
                    className={`
                      max-w-96 space-y-6 text-center

                      lg:text-start
                    `}
                >
                    <div
                        aria-label="logo"
                        className={`
                          inline rounded-md bg-gradient-to-r from-green-400
                          to-blue-500 px-3 text-3xl text-white
                        `}
                    >
                        {'jsonthing'}
                    </div>

                    <h1
                        aria-label=""
                        className={`
                          text-4xl font-bold

                          md:text-5xl
                        `}
                    >
                        {'JSON storage that speaks API.'}
                    </h1>

                    <p className="text-xl text-gray-600">
                        {`Edit,
                        and access your JSON documents with automatically generated API endpoints.`}
                    </p>

                    <CreateNewDocButton className="px-8 py-6 text-base">
                        {'Create new JSON doc'}
                    </CreateNewDocButton>
                </div>

                <div>
                    <Image
                        className="rounded-md"
                        src="https://storage.googleapis.com/jsonthing-assets/hero.jpeg"
                        alt="jsonthing.com"
                        width={600}
                        height={500}
                    />
                </div>
            </main>
        </div>
    )
}
