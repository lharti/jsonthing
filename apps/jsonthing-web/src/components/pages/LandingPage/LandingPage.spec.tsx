import { render } from '@testing-library/react'
import React from 'react'
import { LandingPage } from './LandingPage'

jest.mock('@/components/CreateNewDocButton')

describe('<LandingPage />', () => {
    it('should render', () => {
        expect.assertions(1)

        const { container } = render(<LandingPage />)

        expect(container).toMatchSnapshot('<LandingPage />')
    })
})
