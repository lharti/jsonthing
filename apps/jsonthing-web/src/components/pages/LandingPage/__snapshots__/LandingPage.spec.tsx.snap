// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> should render: <LandingPage /> 1`] = `
<div>
  <div
    class="mx-auto max-w-screen-xl px-8 pt-32"
  >
    <main
      class="
                  flex flex-col items-center gap-8

                  lg:flex-row lg:items-start lg:justify-between
                "
    >
      <div
        class="
                      max-w-96 space-y-6 text-center

                      lg:text-start
                    "
      >
        <div
          aria-label="logo"
          class="
                          inline rounded-md bg-gradient-to-r from-green-400
                          to-blue-500 px-3 text-3xl text-white
                        "
        >
          jsonthing
        </div>
        <h1
          aria-label=""
          class="
                          text-4xl font-bold

                          md:text-5xl
                        "
        >
          JSON storage that speaks API.
        </h1>
        <p
          class="text-xl text-gray-600"
        >
          Edit,
                        and access your JSON documents with automatically generated API endpoints.
        </p>
      </div>
      <div>
        <img
          alt="jsonthing.com"
          class="rounded-md"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fstorage.googleapis.com%2Fjsonthing-assets%2Fhero.jpeg&w=1200&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fstorage.googleapis.com%2Fjsonthing-assets%2Fhero.jpeg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fstorage.googleapis.com%2Fjsonthing-assets%2Fhero.jpeg&w=1200&q=75 2x"
          style="color: transparent;"
          width="600"
        />
      </div>
    </main>
  </div>
</div>
`;
